packages/core/js-client/src/api.ts

```diff
@@ -15,182 +15,158 @@
  */
 
 import type {
-  FnConfig,
   FunctionCallDef,
+  JSONValue,
+  SimpleTypes,
+  ArrowWithoutCallbacks,
   ServiceDef,
-  PassedArgs,
-  ServiceImpl,
 } from "@fluencelabs/interfaces";
-import { getArgumentTypes } from "@fluencelabs/interfaces";
 
+import { CallAquaFunctionConfig } from "./compilerSupport/callFunction.js";
+import {
+  aqua2ts,
+  ts2aqua,
+  wrapFunction,
+  wrapServiceFunction,
+} from "./compilerSupport/conversions.js";
+import { ServiceImpl } from "./compilerSupport/types.js";
 import { FluencePeer } from "./jsPeer/FluencePeer.js";
 
 import { callAquaFunction, Fluence, registerService } from "./index.js";
 
-export const isFluencePeer = (
-  fluencePeerCandidate: unknown,
-): fluencePeerCandidate is FluencePeer => {
-  return fluencePeerCandidate instanceof FluencePeer;
-};
-
 /**
  * Convenience function to support Aqua `func` generation backend
  * The compiler only need to generate a call the function and provide the corresponding definitions and the air script
  *
- * @param rawFnArgs - raw arguments passed by user to the generated function
+ * @param args - raw arguments passed by user to the generated function
  * @param def - function definition generated by the Aqua compiler
  * @param script - air script with function execution logic generated by the Aqua compiler
  */
 export const v5_callFunction = async (
-  rawFnArgs: unknown[],
+  args: (JSONValue | ((...args: JSONValue[]) => JSONValue))[],
   def: FunctionCallDef,
   script: string,
 ): Promise<unknown> => {
-  const { args, client: peer, config } = extractFunctionArgs(rawFnArgs, def);
-
-  return callAquaFunction({
-    args,
-    def,
-    script,
-    config,
-    peer,
-  });
-};
+  const argNames = Object.keys(def.arrow);
+  const argCount = argNames.length;
 
-/**
- * Convenience function to support Aqua `service` generation backend
- * The compiler only need to generate a call the function and provide the corresponding definitions and the air script
- * @param args - raw arguments passed by user to the generated function
- * TODO: dont forget to add jsdoc for new arg
- * @param def - service definition generated by the Aqua compiler
- */
-export const v5_registerService = (args: unknown[], def: ServiceDef): void => {
-  // TODO: Support this in aqua-to-js package
-  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-  const service: ServiceImpl = args.pop() as ServiceImpl;
+  const functionArgs: Record<string, SimpleTypes | ArrowWithoutCallbacks> =
```

Comment:
Is it `functionArgs` or is it `functionArgsSchema`? Seems like a schema to me. Gotta separate values from schema/types, otherwise code is hard to understand.